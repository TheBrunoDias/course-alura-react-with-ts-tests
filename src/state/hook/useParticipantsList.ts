import { participantsListState } from './../atom';
import { useRecoilValue } from 'recoil';

export const useParticipantList = () => {
  return useRecoilValue(participantsListState);
};
