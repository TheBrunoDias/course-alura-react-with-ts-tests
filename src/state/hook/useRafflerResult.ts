import { resultRaffle } from './../atom';
import { useRecoilValue } from 'recoil';

export const useRafflerResult = () => {
  return useRecoilValue(resultRaffle);
};
