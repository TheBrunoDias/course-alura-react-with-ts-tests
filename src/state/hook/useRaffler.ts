import { useSetRecoilState } from 'recoil';
import { doRaffler } from '../helpers/doRaffler';
import { resultRaffle } from './../atom';
import { useParticipantList } from './useParticipantsList';

export const useRaffler = () => {
  const participants = useParticipantList();

  const setResult = useSetRecoilState(resultRaffle);

  return () => {
    const result = doRaffler(participants);
    setResult(result);
  };
};
