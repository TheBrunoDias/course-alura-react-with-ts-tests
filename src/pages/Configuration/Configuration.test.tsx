import { render } from '@testing-library/react';
import React from 'react';
import { RecoilRoot } from 'recoil';
import { Configuration } from '.';
import { useParticipantList } from '../../state/hook/useParticipantsList';
const mockUseNavigation = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockUseNavigation,
  };
});

jest.mock('../../state/hook/useParticipantsList', () => {
  return {
    useParticipantList: jest.fn(),
  };
});
describe('Configuration Page', () => {
  beforeEach(() => {
    (useParticipantList as jest.Mock).mockReturnValue([]);
  });

  test('most be render', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuration />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });
});
