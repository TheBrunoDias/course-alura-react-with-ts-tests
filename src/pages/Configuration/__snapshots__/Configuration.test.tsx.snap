// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration Page most be render 1`] = `
<div>
  <form>
    <input
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Adicionar
    </button>
    
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Come√ßar
    </button>
  </footer>
</div>
`;
